ports/esp32/machine_uart.c

```diff
@@ -48,12 +48,17 @@ typedef struct _machine_uart_obj_t {
     int8_t cts;
     uint16_t txbuf;
     uint16_t rxbuf;
-    uint16_t timeout;       // timeout waiting for first char (in ms)
-    uint16_t timeout_char;  // timeout waiting between chars (in ms)
-    uint32_t invert;        // lines to invert
+    uint16_t timeout;             // timeout waiting for first char (in ms)
+    uint16_t timeout_char;        // timeout waiting between chars (in ms)
+    uint32_t invert;              // lines to invert
+    uint8_t uart_mode;            // UART mode
+    uint8_t flow_ctrl;            // hardware flow control
+    uint8_t rx_flow_ctrl_thresh;  // hardware flow control RTS threshold
 } machine_uart_obj_t;
 
 STATIC const char *_parity_name[] = {"None", "1", "0"};
+STATIC const char *_uart_mode_name[] = {"UART", "RS485_HALF_DUPLEX", "IRDA", "RS485_COLLISION_DETECT", "RS485_APP_CTRL"};
+STATIC const char *_flow_ctrl_name[] = {"DISABLE", "RTS", "CTS", "CTS_RTS"};
```

Comment:
These should be python constants on `machine.UART` IMHO.

Reply:
Thanks for comment @tve.
I am not expert of Micropython. Any pointer to help me implementing a python constant?

Reply:
For example https://github.com/micropython/micropython/blob/master/ports/esp32/machine_uart.c#L383-L386